src/pages/RFOX/hooks/useAccountLogsQuery.ts

```diff
@@ -0,0 +1,85 @@
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
+import { useMemo } from 'react'
+import { getAddress } from 'viem'
+import { arbitrum } from 'viem/chains'
+import { viemClientByNetworkId } from 'lib/viem-client'
+
+import { setRuneAddressEvent, stakeEvent, unstakeEvent, withdrawEvent } from '../constants'
+import type { RFOXAccountLog } from '../types'
+import { getRfoxContractCreationBlockNumber } from './helpers'
+
+type UseAccountLogsProps<SelectData> = {
+  stakingAssetAccountAddress: string | undefined
+  select?: (sortedAccountLogs: RFOXAccountLog[]) => SelectData
+}
+
+const client = viemClientByNetworkId[arbitrum.id]
+
+/**
+ * Calculates the time the account has most recently had a non-zero staking balance to now, in seconds
+ *
+ * WARNING: This is a very expensive operation and should only be used behind react-query to cache the result.
+ *
+ * i.e DO NOT EXPORT THIS FUNCTION
+ */
+const fetchAccountLogs = async (stakingAssetAccountAddress: string): Promise<RFOXAccountLog[]> => {
```

Comment:
yeah so uh.. I implemented this to re-use for grabbing rune addresses corresponding to the account, but remembered that rune addresses on logs are actually keccak hashes, not strings. Left it here because it naice.